crates/client/src/client/types/gas_costs.rs

```diff
@@ -0,0 +1,345 @@
+use crate::client::schema;
+
+pub struct GasCosts {
```

Comment:
You can create a local macro and automate `From` implementation=) We did the same [here](https://github.com/FuelLabs/fuel-vm/pull/499/files#diff-b5eeedfa720b67c4498e4b6d75e8bf7019ee7d60494d7511f86854118566b888)

Reply:
Would you prefer I do a macro? I just let copilot do the work for me :).

Reply:
A macro might be easier to maintain as we add new fields in the future

Reply:
This case is a bit different from the example you gave since it's a struct instead of an enum. I believe it is still possible to create a proc macro that would iterate over all the fields (see: https://github.com/dtolnay/syn/issues/516). Idk if I'm convinced it's worth my time though--I don't mind the current verbosity.

Reply:
You still can do it with `macro_rules` without proc macro=)

Reply:
Done.